### PR TITLE
Join suggestions and error span

### PR DESCRIPTION
**Pull request by anderson-joyle**
-  Fixing span information for error messages.
- Scope fields suggestions.

### CODE REVIEW

**Review by MikeStall — DISMISSED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/ArgumentSuggestions.cs [hunk 1]
```diff
@@ -54,6 +55,12 @@ public static IEnumerable<KeyValuePair<string, DType>> GetArgumentSuggestions(In
                 return suggestor(tryGetEnumSymbol, suggestUnqualifiedEnums, scopeType, argumentIndex, out requiresSuggestionEscaping);
             }
 
+            // !!! This block will not be necessary once we move Join to Core
+            if (function is JoinFunction joinFunction)
+            {
+                return JoinSuggestion(tryGetEnumSymbol, suggestUnqualifiedEnums, scopeType, argumentIndex, out requiresSuggestionEscaping);
+            }
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/ArgumentSuggestions.cs:62]: @MikeStall

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/ArgumentSuggestions.cs (hunk 1 above)
Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/ArgumentSuggestions.cs:62]: Consider adding a new virtual to TexlFunction

virtual Type DeclarationType => this.GetType();

And then joinImpl can override:
  override Type DeclarationType => typeof(JoinFunction);

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**